src/main/java/com/google/research/bleth/simulator/StatisticsState.java

```diff
@@ -135,9 +185,22 @@ public void writeBeaconsObservedStats() {
     }
 
     private StatisticsState(String simulationId, Map<String, Double> distanceStats,
-                            Table<String, String, Double> beaconsObservedStats) {
+                            Table<String, String, Double> beaconsObservedStats,
+                            LinkedListMultimap<Integer, ObservedInterval> beaconsObservedIntervals) {
         this.simulationId = simulationId;
         this.distanceStats = distanceStats;
         this.beaconsObservedStats = beaconsObservedStats;
+        this.beaconsObservedIntervals = beaconsObservedIntervals;
+    }
+
+    private static ObservedInterval extractObservedInterval(Entity entity) {
+        int start = ((Long) entity.getProperty(Schema.StatisticsState.intervalStart)).intValue();
```

Comment:
Do you really need this trip through `Long` --> `intValue()`? Couldn't you just cast directly to `Integer`?

Reply:
According to [setProperty](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyContainer.html#setProperty-java.lang.String-java.lang.Object-) documentation:

> As the value is stored in the datastore, it is converted to the datastore's native type. This may include widening, such as converting a Short to a Long. 

Also discussed [here](https://stackoverflow.com/questions/15246265/calling-setproperty-with-int-but-getproperty-returns-long-on-google-app-engine-p).